### ISSUE TITLE
`_changes` filtered on `_doc_ids` returns unexpected `last_seq`

### ISSUE THREAD

**Issue by kdoran** [apache/couchdb#1221]
## Expected Behavior
From: http://docs.couchdb.org/en/2.1.1/api/database/changes.html#post--db-_changes:

> "last_seq is the update sequence of the last update returned (Equivalent to the last item in the results)."

When filtering on `_doc_ids`, I was expecting `last_seq` to be the same thing as the last `seq` in the list of changes returned when filtering on _doc_ids.

## Current Behavior
When filtering on `_doc_ids`, `_changes` responds with the latest sequence in the database or, if there's a `limit` argument, the sequence number after the limit number.

## Steps to Reproduce (for bugs)
1. `POST /{db}/_changes?filter=_doc_ids&since=0` with body `{doc_ids: ['one-example-id']}`
2. See that `last_seq` in response is not the last `seq` in the list of changes.

## Context
Grabbing the `last_seq` from the `_changes` response and using it as the `since` parameter on the next request to `_changes` is PouchDB's approach to a replication filtered on IDs. Using that later sequence number means documents are (sometimes) missed in the replication. 

https://github.com/pouchdb/pouchdb/issues/6809

## Your Environment
Saw this when running a single node CouchDB 2.1 & on CouchDB 1.6 locally on a macbook pro.

**Comment by wohali** [comment]
@kdoran Could this be a duplicate of #748?

**Comment by kdoran** [comment]
@wohali it seemed like it could be, but then I saw the same behavior happening on a 1.7x... so would that mean it's a separate issue?

**Comment by wohali** [comment]
@kdoran Definitely worth investigating, I missed that it was also an issue in 1.x.

**Comment by kdoran** [comment]
👍  for our case it ended up being just fine to use the last `seq` in the change results list instead of `last_seq` in case others stumble across this.

**Comment by wohali** [comment]
@kdoran We need a `last_seq` and the final row's `seq` value to investigate further, of a response that has caused the issue. Can you provide? Thanks!

**Comment by wohali** [comment]
I wonder if this is related to #984 and @rnewson 's investigation there.

**Comment by kdoran** [comment]
@wohali yup, sorry for the delay! Here's a quick js example I ran in the chrome devtools console while on the fauxton page on my local CouchDB, which is 2.1.0 as a single node. The same discrepancy between `seq` and `last_seq` also happens when running this on a remote EC2 CouchDB running 2.1.1 also as a single node:

```
couchUrl = 'http://localhost:5984/'
dbName = 'sync-test'
promise = fetch(couchUrl + dbName + '/_all_docs?limit=1&descending=true', 
  {
    headers: {'Content-Type': 'application/json'},
    credentials: 'include'
  }
).then(resp => resp.json()).then(response => response.rows[0].id)
.then(doc_id => {
  return fetch(couchUrl + dbName + '/_changes?limit=10&filter=_doc_ids', 
    {
      method: 'POST', 
      body: JSON.stringify({ doc_ids: [ doc_id ] }),
      headers: {'Content-Type': 'application/json'}
    }
  ).then(resp => resp.json()).then(resp => JSON.stringify(resp, null, 2))
})
Promise.resolve(
  promise
).then(console.log)
```

the response:

```
{
  "results": [
    {
      "seq": "94-g1AAAAB5eJzLYWBgYMpgTmEQTM4vTc5ISXLIyU9OzMnILy7JAUklMiTV____PyvRGI-iPBYgydAApP6D1GYwJ2rnAgXYU1MtUsyNjLDpywIABfcnAA",
      "id": "9::remote",
      "changes": [
        {
          "rev": "1-967a00dff5e02add41819138abb3284d"
        }
      ]
    }
  ],
  "last_seq": "128-g1AAAACbeJzLYWBgYMpgTmEQTM4vTc5ISXLIyU9OzMnILy7JAUklMiTV____PyuDOdEnFyjAnmaanGZsmoRNAx5j8liAJEMDkPoPNc0EbFpqqkWKuZERNn1ZALTyMYU",
  "pending": 0
}
```

and here is the result when I run that same snippet on my local 1.6.1 CouchDB:

```
{
  "results": [
    {
      "seq": 9,
      "id": "9::remote",
      "changes": [
        {
          "rev": "1-967a00dff5e02add41819138abb3284d"
        }
      ]
    }
  ],
  "last_seq": 1019
}
```

**Comment by janl** [comment]
This test suggests that CouchDB is doing the right thing: https://github.com/apache/couchdb/blob/d36149ec3a0d071d067589c8f3fe7d263626948d/src/couch/test/couch_changes_tests.erl#L178-L195

That means we need more information about the specific CouchDB database and server/cluster configuration before we can fix this and amend the test accordingly.

Can somebody here provide us with:
1. a database that exhibits this behaviour, any of these forms are acceptable: .couch file, live read-only(!) database, script in a. common programming language or curl statements that create the database.
2. the exact query that shows the behaviour
3. the server/cluster configuration where that happens. Specifically important are q&n values, number of nodes in the cluster, etc. but anything really you can think of.

Closing this until the above is provided
